lib/mayaUsd/fileio/utils/meshWriteUtils.cpp

```diff
@@ -477,4 +478,76 @@ UsdMayaMeshUtil::assignSubDivTagsToUSDPrim(MFnMesh& meshFn,
     }
 }
 
+void
+UsdMayaMeshUtil::writeVertexData(const MFnMesh& meshFn,
+                                 UsdGeomMesh& primSchema,
+                                 const UsdTimeCode& usdTime,
+                                 UsdUtilsSparseValueWriter& valueWriter)
+{
+    MStatus status{MS::kSuccess};
+
+    const uint32_t numVertices = meshFn.numVertices();
```

Comment:
Looks like the return type here should is just `int`?

Reply:
Correct, the return type of numVertices() is `int`. Both `int `and `uint32_t `are `4 bytes` in size so we should be Ok here.

Reply:
Right, the size is the same, but the semantics between the two types are not. I'd argue that `numVertices()` **should** return an unsigned type, but it doesn't. I usually prefer to store the returned value using the type that matches exactly, then do some kind of explicit casting if necessary, rather than introduce an implicit conversion.

Reply:
I am on the same boat as you on this.